src/apps/dolomite/arbitrum/dolomite.contract-position.template.position-fetcher.ts

```diff
@@ -0,0 +1,327 @@
+import { Inject } from '@nestjs/common';
+import { BigNumber, BigNumberish } from 'ethers';
+import { gql, GraphQLClient } from 'graphql-request';
+import { compact, sumBy } from 'lodash';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import { drillBalance } from '~app-toolkit/helpers/drill-balance.helper';
+import { DolomiteContractFactory, DolomiteMargin } from '~apps/dolomite/contracts';
+import { CHUNK_SIZE, DOLOMITE_GRAPH_ENDPOINT, DOLOMITE_MARGIN_ADDRESSES } from '~apps/dolomite/utils';
+import {
+  AccountStruct,
+  DolomiteContractPosition,
+  DolomiteDataProps,
+  DolomiteTokenDefinition,
+  getAllIsolationModeTokensFromContractPositions,
+  getTokenBalancesPerAccountStructLib,
+  getTokenDefinitionsLib,
+  mapTokensToDolomiteDataProps,
+} from '~apps/dolomite/utils';
+import { IMulticallWrapper } from '~multicall';
+import { ContractType } from '~position/contract.interface';
+import { ContractPositionBalance } from '~position/position-balance.interface';
+import { MetaType } from '~position/position.interface';
+import { metatyped } from '~position/position.utils';
+import {
+  DefaultContractPositionDefinition,
+  GetDataPropsParams,
+  GetDefinitionsParams,
+  GetDisplayPropsParams,
+  GetTokenBalancesParams,
+  GetTokenDefinitionsParams,
+} from '~position/template/contract-position.template.types';
+import { CustomContractPositionTemplatePositionFetcher } from '~position/template/custom-contract-position.template.position-fetcher';
+
+export abstract class DolomiteContractPositionTemplatePositionFetcher extends CustomContractPositionTemplatePositionFetcher<
+  DolomiteMargin,
+  DolomiteDataProps
+> {
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(DolomiteContractFactory) protected readonly dolomiteContractFactory: DolomiteContractFactory,
+  ) {
+    super(appToolkit);
+  }
+
+  protected abstract isFetchingDolomiteBalances(): boolean;
+  getContract(address: string): DolomiteMargin {
+    return this.dolomiteContractFactory.dolomiteMargin({ address, network: this.network });
+  }
+
+  getDefinitions(_params: GetDefinitionsParams): Promise<DefaultContractPositionDefinition[]> {
+    return Promise.resolve([
+      {
+        address: DOLOMITE_MARGIN_ADDRESSES[this.network],
+      },
+    ]);
+  }
+
+  async getTokenDefinitions(
+    params: GetTokenDefinitionsParams<DolomiteMargin>,
+  ): Promise<DolomiteTokenDefinition[] | null> {
+    return getTokenDefinitionsLib(params, this.dolomiteContractFactory, this.network);
+  }
+  override async getPositions() {
```

Comment:
You shouldn't have to override this. I've read the comments, but I don't really understand

Reply:
See the new comments I put underneath `getPositions` on line 66 to better understand. That comment about the token loader became out of date.

Reply:
I understand your point, but I still wouldn't override `getPositions()` for main reason it will become a pain to maintain especially if we update templates. A few RPC calls won't make much of a difference compared to the amount we're already making